websites/Z/ZenMarket/presence.ts

```diff
@@ -17,249 +17,256 @@ presence.on("UpdateData", async () => {
-		presenceData.buttons = [
-			{
-				label: "View on ZenMarket",
-				url: href,
-			},
-			{
-				label: "View on Rakuten",
-				url: document.querySelector("#productPage").getAttribute("href"),
-			},
-		];
-	}
+				presenceData.state = text;
+				if (!item?.productPage && !item?.url) {
+					presenceData.buttons = [
+						{
+							label: "View On ZenMarket",
+							url: href,
+						},
+					];
+					return presence.setActivity(presenceData);
+				}
 
-	if (path === "/amazonproduct.aspx") {
-		const condition: HTMLSpanElement =
-			document.querySelector("#lblConditionName");
-		presenceData.details = `Looking at ${
-			document.querySelector("#itemTitle").textContent
-		}`;
-		presenceData.state = `Price: ${
-			document.querySelector("#lblPrice").textContent
-		} ${condition ? `|| Condition: ${condition.textContent}` : ""}`;
-		presenceData.buttons = [
-			{
-				label: "View on ZenMarket",
-				url: href,
-			},
-			{
-				label: "View on Amazon",
-				url: document.querySelector("#productPage").getAttribute("href"),
-			},
-		];
-	}
+				switch (
+					document.querySelector(".active-tab")?.textContent?.toLowerCase()
+				) {
+					case "amazon": {
+						presenceData.buttons = [
+							{
+								label: "View On ZenMarket",
+								url: href,
+							},
+							{
+								label: "View On Amazon",
+								url: item?.productPage,
+							},
+						];
+						break;
+					}
+					case "rakuten": {
+						presenceData.buttons = [
+							{
+								label: "View On ZenMarket",
+								url: href,
+							},
+							{
+								label: "View On Rakuten",
+								url: item?.productPage,
+							},
+						];
+						break;
+					}
+					case "zenplus": {
+						presenceData.buttons = [
+							{
+								label: "View On ZenMarket",
+								url: href,
+							},
+							{
+								label: "View On ZenPlus",
+								url: item?.productPage,
+							},
+						];
+						break;
+					}
+					case "y! shopping": {
+						presenceData.buttons = [
+							{
+								label: "View On ZenMarket",
+								url: href,
+							},
+							{
+								label: "View On Yahoo Shopping",
+								url: item?.productPage,
+							},
+						];
+						break;
+					}
+					case "mercari": {
+						presenceData.buttons = [
+							{
+								label: "View On ZenMarket",
+								url: href,
+							},
+							{
+								label: "View On Mercari",
+								url: item?.productPage,
+							},
+						];
+						break;
+					}
+					case "rakuten rakuma": {
+						presenceData.buttons = [
+							{
+								label: "View On ZenMarket",
+								url: href,
+							},
+							{
+								label: "View On Rakuma",
+								url: item?.productPage,
+							},
+						];
+						break;
+					}
+					default: {
+						presenceData.buttons = [
+							{
+								label: "View On ZenMarket",
+								url: href,
+							},
+							{
+								label: "View On Other Market",
+								url: item?.productPage,
+							},
+						];
+					}
+				}
```

Comment:
Would love to shorten this with a dynamic button name ;)

Reply:
If @theusaf agrees you can IMO.

Reply:
I think that's fine.